aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -690,47 +698,43 @@ AT_ERROR("inverse: MAGMA library not found in "
 }
 
 template <typename scalar_t>
-static void apply_single_inverse(Tensor& self, int64_t& info) {
-#ifndef USE_MAGMA
-AT_ERROR("inverse: MAGMA library not found in "
-    "compilation. Please rebuild with MAGMA.");
-#else
+static void apply_single_inverse(const Tensor& self, int64_t& info, Tensor& ret) {
   auto self_data = self.data_ptr<scalar_t>();
-  magma_int_t n = magma_int_cast(self.size(-2), "self.size(-2)");
-  magma_int_t lwork = n * magmaGetriOptimalBlocksize<scalar_t>(n);
-  magma_int_t info_tmp = 0;
+  int n = magma_int_cast(self.size(-2), "self.size(-2)");
+  int info_tmp = 0;
 
-  Tensor ipiv = at::empty({n}, at::kInt);
-  Tensor dwork = at::empty({lwork}, self.options());
-  magmaLu<scalar_t>(n, n, self_data, n, ipiv.data_ptr<magma_int_t>(), &info_tmp);
+  Tensor ipiv = at::empty({n}, self.options().dtype(at::kInt));
+  cu_xgetrf<scalar_t>(n, n, self_data, n, ipiv.data_ptr<int>(), &info_tmp);
   if (info_tmp != 0) {
     info = info_tmp;
     return;
   }
-  magmaGetri<scalar_t>(
-    n, self_data, n, ipiv.data_ptr<magma_int_t>(), dwork.data_ptr<scalar_t>(), lwork, &info_tmp);
+  ret = at::eye(n, self.options());
```

Comment:
Why initializing it as `eye`?

Reply:
cusolver doesn't have xgetri, it only has xgetrs, where s is for solve. We let rhs of AX=B be identity matrix, then this will be equivalent to inverse.